rpd_generator/bdl_structure/bdl_commands/zone.py

```diff
@@ -1068,3 +1102,160 @@ def calculate_fan_power(self, airflow, pressure_rise, total_efficiency):
+                occ_cfm_allows_dcv_to_take_effect = True
+
+        min_oa_sch = self.parent.get_obj(
+            self.parent.get_inp(BDL_SystemKeywords.MIN_AIR_SCH)
+        )
+        fan_sch = self.parent.get_obj(
+            self.parent.get_inp(BDL_SystemKeywords.FAN_SCHEDULE)
+        )
+        dcv_flow_control_options = [
+            BDL_MinFlowControlOptions.DCV_RESET_DOWN,
+            BDL_MinFlowControlOptions.DCV_RESET_UP_DOWN,
+        ]
+
+        if min_oa_sch is not None and (
+            not zone_is_attached_to_sys_with_terminal_inputs
+            or (
+                zone_is_attached_to_sys_with_terminal_inputs
+                and self.get_inp(BDL_ZoneKeywords.MIN_FLOW_CTRL)
+                not in dcv_flow_control_options
+            )
+        ):
+            dcv_never_takes_precedence_due_to_min_air_sch = (
+                False
+                if min_oa_sch is None
+                else not any(
+                    min_oa_sch.hourly_values[i] == -999
+                    for i in range(len(fan_sch.hourly_values))
+                    if fan_sch.hourly_values[i] == 1
+                )
+            )
+        else:
+            dcv_never_takes_precedence_due_to_min_air_sch = False
+
+        # For units with terminal inputs ,
+        # their hourly specified values will always be used, for the appropriate VAV box minimum flow setting
+        min_oa_method = self.parent.get_inp(BDL_SystemKeywords.MIN_OA_METHOD)
+
+        if (
+            zone_is_attached_to_sys_with_terminal_inputs
+            and self.get_inp(BDL_ZoneKeywords.MIN_FLOW_CTRL) in dcv_flow_control_options
+        ):
+            terminal_flow_schedules = [
+                sch
+                for sch in [
+                    self.get_obj(self.get_inp(BDL_ZoneKeywords.MIN_FLOW_SCH)),
+                    self.get_obj(self.get_inp(BDL_ZoneKeywords.CMIN_FLOW_SCH)),
+                    self.get_obj(self.get_inp(BDL_ZoneKeywords.HMIN_FLOW_SCH)),
+                ]
+                if sch is not None
+            ]
+
+            # if MIN-FLOW-SCH, CMIN-FLOW-SCH or HMIN-FLOW-SCH are specified
+            if any(terminal_flow_schedules):
+
+                # check if hourly value is -999 for all flow schedules while the fan schedule value is 1
+                for i in range(len(fan_sch.hourly_values)):
+
+                    if fan_sch.hourly_values[i] == 1:
+                        if all(
+                            lst.hourly_values[i] == -999
+                            for lst in terminal_flow_schedules
+                        ) or min_oa_method in [
+                            BDL_SystemMinimumOutdoorAirControlOptions.DCV_RETURN_SENSOR,
+                            BDL_SystemMinimumOutdoorAirControlOptions.DCV_ZONE_SENSORS,
+                        ]:
+                            dcv_never_takes_precedence_due_to_min_air_sch = False
+                            break
+
+                    if fan_sch is None:
```

Comment:
this check `if fan_sch is None:` is happening inside of an iteration through `fan_sch.hourly_values`. Can you help me understand the intent?